packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -322,22 +397,120 @@ class TimelineEvent {
   TimelineEvent parent;
   List<TimelineEvent> children = <TimelineEvent>[];
 
-  int get duration => endTime - startTime;
-
-  bool get wellFormed => startTime != null && duration != null;
+  int get duration => (endTime != null) ? endTime - startTime : null;
 
   bool get isCpuEvent => type == TimelineEventType.cpu;
   bool get isGpuEvent => type == TimelineEventType.gpu;
 
+  bool get isPipelineProduceFlow => _hasChild('Engine::BeginFrame');
+  bool get isPipelineConsumeFlow => _hasChild('PipelineConsume');
+
+  /// Whether there is a child with the given name [childName] is contained
+  /// somewhere in the subtree [children].
+  bool _hasChild(String childName) {
+    bool childFound = false;
+
+    void findChild(TimelineEvent event, String childName) {
+      if (event.name.contains(childName)) {
+        childFound = true;
+      }
+      if (childFound) return;
+      for (TimelineEvent e in event.children) {
+        findChild(e, childName);
+      }
+    }
+
+    findChild(this, childName);
+    return childFound;
+  }
+
+  /// Adds a child event.
+  ///
+  /// If we know the child's location (knownLocation == true)
```

Comment:
This looks like a comment that should go in the method body instead of as a doc comment as it is describing the implementation instead of describing things I should know to understand the API.

Reply:
removed comment entirely.